src/codegen/array_boundary.rs

```diff
@@ -0,0 +1,159 @@
+use super::vartable::Vartable;
+use crate::ast;
+use crate::codegen::cfg::{ControlFlowGraph, Instr};
+use crate::codegen::Expression;
+use crate::sema::ast::Type;
+use indexmap::IndexMap;
+use num_bigint::BigInt;
+use num_traits::One;
+use solang_parser::pt::Loc;
+
+// IndexMap <ArrayVariable res , res of temp variable>
+pub type ArrayTempVars = IndexMap<usize, usize>;
+
+// Function to modify arraye length temp
+pub(crate) fn modify_temp_array_size(
+    cfg: &mut ControlFlowGraph,
+    minus: bool,        // If the function is called from pushMemory or popMemory
+    address_arr: usize, // The res of array that push/pop is performed on
+    vartab: &mut Vartable,
+) {
+    // If not empty
+    if let Some(to_add) = cfg.array_lengths_temps.clone().get(&address_arr) {
```

Comment:
Don't clone the data structure. You only need a single `usize`, but you are cloning an entire HashMap to retrieve it!

Reply:
I was trying to avoid an immutable borrow here, because a mutable borrow is later used in cfg.add().  Is there a way to get an immutable copy without cloning?

Reply:
> I was trying to avoid an immutable borrow here, because a mutable borrow is later used in cfg.add(). Is there a way to get an immutable copy without cloning?

If you are sure there will be a corresponding value for your `address_arr` key, we can retrieve it without an if-else condition by doing `let to_add = cfg.array_lengths_temps[&address_arr]`. If the condition is needed, you need to create first and if-else that retrieves the value and then another one to create the expression.